Fedora Rawhide got compilation issue in CI

Bug Report
==========
<!--
Please fill in a a meaningful and concise title in the field above.
It helps us to classify the issue even  before reading the body and handling the
issue a bit better.

Examples:
  * Great: "Unable to select fleet in galaxy map"
  * Good: "Unable to click fleet"
  * Not really okay: "Issue with fleets"
  * Horrible: "HELP!!!"
-->


Environment
-----------
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example a crashing hosted server may not need
information about the graphic card, but maybe depend on the operating system you
use.  When in doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**: master
* **Operating System**: Linux Fedora Rawhide
* **Fetched as** <!-- delete lines below that don't apply -->
  * Compiled from source


Description
-----------
<!--
Add a meaningful description of the bug you encountered.

If a screenshot or image helps to describe the issue content feel free to create
one.

Also attach the log files the game creates.  The log files The log files can be
found on
* Windows: %APPDATA%\FreeOrion
* MacOSX: ~/Library/"Application Support"/FreeOrion
* Linux: ${XDG_DATA_HOME:-~/.local/share}/freeorion
-->

https://github.com/freeorion/freeorion/actions/runs/4413923912/jobs/7738343348

```
...
[ 60%] Linking CXX shared library libfreeorionparse.so
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x0): multiple definition of `boost::phoenix::placeholders::uarg10'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x2): multiple definition of `boost::phoenix::placeholders::uarg9'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x2): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x4): multiple definition of `boost::phoenix::placeholders::uarg8'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x6): multiple definition of `boost::phoenix::placeholders::uarg7'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x6): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x8): multiple definition of `boost::phoenix::placeholders::uarg6'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x8): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0xa): multiple definition of `boost::phoenix::placeholders::uarg5'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0xa): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0xc): multiple definition of `boost::phoenix::placeholders::uarg4'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0xc): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0xe): multiple definition of `boost::phoenix::placeholders::uarg3'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0xe): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x10): multiple definition of `boost::phoenix::placeholders::uarg2'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x10): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o:(.bss+0x12): multiple definition of `boost::phoenix::placeholders::uarg1'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x12): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0x0): multiple definition of `boost::phoenix::placeholders::uarg10'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x0): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0x2): multiple definition of `boost::phoenix::placeholders::uarg9'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x2): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0x4): multiple definition of `boost::phoenix::placeholders::uarg8'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x4): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0x6): multiple definition of `boost::phoenix::placeholders::uarg7'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x6): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0x8): multiple definition of `boost::phoenix::placeholders::uarg6'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0x8): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0xa): multiple definition of `boost::phoenix::placeholders::uarg5'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0xa): first defined here
/usr/bin/ld: CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o:(.bss+0xc): multiple definition of `boost::phoenix::placeholders::uarg4'; CMakeFiles/freeorionparseobj.dir/parse/ArithmeticRules.cpp.o:(.bss+0xc): first defined here
...
```


Expected Result
---------------
<!--
When you enter an issue please add a description of what behaviour you would
expect from the game instead of the issue.
-->
Successful compilation.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,6 +346,8 @@ set_property(DIRECTORY APPEND
         $<$<PLATFORM_ID:FreeBSD>:BOOST_DATE_TIME_NO_LOCALE>
         # Forcing backtrace using api for boost stacktrace due to lacking on ::Unwind side
         $<$<PLATFORM_ID:FreeBSD>:BOOST_STACKTRACE_USE_LIBC_BACKTRACE_FUNCTION>
+        # Workaround for boost 1.81 issue https://github.com/boostorg/phoenix/issues/111
+        $<$<VERSION_EQUAL:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.81>:BOOST_PHOENIX_STL_TUPLE_H_>
 )
 
 

